crates/oxc_transformer/src/plugins/styled_components.rs

```diff
@@ -1062,20 +1062,36 @@ fn minify_template_literal<'a>(lit: &mut TemplateLiteral<'a>, ast: AstBuilder<'a
                 // Skip and compress whitespace.
                 _ if cur_byte.is_ascii_whitespace() => {
                     i += 1;
-                    // Compress symbols, remove spaces around these symbols,
-                    // but preserve whitespace preceding colon, to avoid joining selectors.
-                    if output.last().is_some_and(|&last| {
-                        !matches!(last, b' ' | b':' | b'{' | b'}' | b',' | b';')
-                    }) && (i == bytes.len() || !matches!(bytes[i], b'{' | b'}' | b',' | b';'))
+                    // Decide whether to preserve this whitespace character.
+                    // CSS allows removing spaces around certain delimiters without changing meaning:
+                    // - `color: red` -> `color:red` (spaces around colons)
+                    // - `.a { }` -> `.a{}` (spaces around braces)
+                    // - `margin: 1px , 2px` -> `margin:1px,2px` (spaces around commas)
+                    // But spaces are significant in other contexts like selectors: `.a .b` != `.a.b`
+                    if output.last().map_or(
+                        // Case 1: If output is empty (no last char), preserve space only if we're
+                        // in a non-first quasi to avoid joining with the previous interpolation.
+                        // Example: `${A} ${B}` - the space between interpolations must be preserved
+                        quasi_index != 0,
+                        // Case 2: If we have a last char, preserve space unless it's a CSS delimiter
+                        // that can safely have adjacent spaces removed (space, colon, braces, comma, semicolon)
+                        |&last| !matches!(last, b' ' | b':' | b'{' | b'}' | b',' | b';'),
+                    )
+                    // AND check what comes after this whitespace:
+                    // - If we're at the end of the quasi (i == bytes.len()), preserve the space
+                    //   to avoid joining with the next interpolation
+                    // - Otherwise, only preserve if the next char is NOT one of: { } , ;
+                    //   Note: We intentionally DON'T include ':' here because spaces before colons
+                    //   are significant in CSS. ` :hover` (descendant pseudo-selector) is different
+                    //   from `:hover` (direct pseudo-selector). Example: `.parent :hover` selects any
+                    //   hovered descendant, while `.parent:hover` selects the parent when hovered.
+                    && (i == bytes.len() || !matches!(bytes[i], b'{' | b'}' | b',' | b';'))
```

Comment:
I'm not completely sure, but could we omit this whitespace character if next char is whitespace too?

```suggestion
                    && bytes.get(i).is_none_or(|&b| {
                        !b.is_ascii_whitespace() && !matches!(b, b'{' | b'}' | b',' | b';')
                    })
```

It would make a difference in a case like `something:     {`.

Reply:
I'm not sure what patterns this would cover exactly. I'll try it out and do a follow-up PR.